lldb/source/Host/aix/HostInfoAIX.cpp

```diff
@@ -0,0 +1,213 @@
+      std::string get_distribution_id_command(get_distribution_info_exe);
+      get_distribution_id_command += " -i";
+
+      FILE *file = popen(get_distribution_id_command.c_str(), "r");
+      if (!file) {
+        LLDB_LOGF(log,
+                  "failed to run command: \"%s\", cannot retrieve "
+                  "platform information",
+                  get_distribution_id_command.c_str());
+        break;
+      }
+
+      // retrieve the distribution id string.
+      char distribution_id[256] = {'\0'};
+      if (fgets(distribution_id, sizeof(distribution_id) - 1, file) !=
+          nullptr) {
+        LLDB_LOGF(log, "distribution id command returned \"%s\"",
+                  distribution_id);
+
+        const char *const distributor_id_key = "Distributor ID:\t";
+        if (strstr(distribution_id, distributor_id_key)) {
+          // strip newlines
+          std::string id_string(distribution_id + strlen(distributor_id_key));
+          llvm::erase(id_string, '\n');
+
+          // lower case it and convert whitespace to underscores
+          std::transform(
+              id_string.begin(), id_string.end(), id_string.begin(),
+              [](char ch) { return tolower(isspace(ch) ? '_' : ch); });
+
+          g_fields->m_distribution_id = id_string;
+          LLDB_LOGF(log, "distribution id set to \"%s\"",
+                    g_fields->m_distribution_id.c_str());
+        } else {
+          LLDB_LOGF(log, "failed to find \"%s\" field in \"%s\"",
+                    distributor_id_key, distribution_id);
+        }
+      } else {
+        LLDB_LOGF(log,
+                  "failed to retrieve distribution id, \"%s\" returned no"
+                  " lines",
+                  get_distribution_id_command.c_str());
+      }
+
+      // clean up the file
+      pclose(file);
+    }
+  });
+
+  return g_fields->m_distribution_id;
+}
+
+FileSpec HostInfoAIX::GetProgramFileSpec() {
+  static FileSpec g_program_filespec;
+
+  if (!g_program_filespec) {
+    char exe_path[PATH_MAX];
+    ssize_t len = readlink("/proc/self/exe", exe_path, sizeof(exe_path) - 1);
+    if (len > 0) {
+      exe_path[len] = 0;
+      g_program_filespec.SetFile(exe_path, FileSpec::Style::native);
+    }
+  }
+
+  return g_program_filespec;
+}
+
+bool HostInfoAIX::ComputeSupportExeDirectory(FileSpec &file_spec) {
+  if (HostInfoPosix::ComputeSupportExeDirectory(file_spec) &&
+      file_spec.IsAbsolute() && FileSystem::Instance().Exists(file_spec))
+    return true;
+  file_spec.SetDirectory(GetProgramFileSpec().GetDirectory());
+  return !file_spec.GetDirectory().IsEmpty();
+}
+
+bool HostInfoAIX::ComputeSystemPluginsDirectory(FileSpec &file_spec) {
+  FileSpec temp_file("/usr/" LLDB_INSTALL_LIBDIR_BASENAME "/lldb/plugins");
+  FileSystem::Instance().Resolve(temp_file);
+  file_spec.SetDirectory(temp_file.GetPath());
+  return true;
+}
+
+bool HostInfoAIX::ComputeUserPluginsDirectory(FileSpec &file_spec) {
```

Comment:
@emaste, @mgorny, @DavidSpickett do you happen to know what's the attitude towards XDG in the BSD land. I'm wondering if we should put this into the base class.

Also, what is the attitude towards XDG in AIX?

Reply:
* https://man.freebsd.org/cgi/man.cgi?query=xdg-user-dirs-update&sektion=1&manpath=freebsd-release-ports exists but only in ports.
* https://www.reddit.com/r/freebsd/comments/1dw59hy/psa_freebsd_141_and_xdg_runtime_dir/ claims that 14.1 exported some XDG variable, which is a positive sign.
* https://github.com/search?q=repo%3Afreebsd%2Ffreebsd-src+%2F%28%3F-i%29XDG_%2F&type=code shows that https://github.com/freebsd/freebsd-src/blob/60f87c736807bd2b5a3ebe451add868232facf43/lib/libpam/modules/pam_xdg/pam_xdg.c#L134 sets up XDG_RUNTIME_DIR, but the rest of the references are in the contrib folders so those could be code meant for Linux.
* A few references in the manual https://docs.freebsd.org/search?DB=en&P=xdg but no clear endorsement.

So FreeBSD itself seems to be ok with XDG, but the user would have to export XDG_DATA_HOME themselves. Or maybe their desktop environment would do that for them?

Reply:
We can say that XDG is used by some people on FreeBSD, and if they do we can assume XDG_DATA_HOME points to something the XDG spec expects it to. So if we find it, using it doesn't seem like a risk to me.

Reply:
Yeah, that makes sense to me. @DhruvSrivastavaX, can you prepare a patch which moves the common parts into HostInfoPosix (by my count, that's GetOSVersion, GetOSBuildString, ComputeSupportExeDirectory, ComputeSystemPluginsDirectory and ComputeUserPluginsDirectory). We can leave this patch for the AIX-specific parts.

Reply:
Yes @labath, I am working on that.